# SQLite Stock Market

This project creates and analyzes a data table including stock market information from IBM Common Stock and Intel Corporation stock during the last 8 days of January 2022 (1/24/2022 through 1/31/2022). The columns include:

`symbol`: the stock abbreviation

`name`: the name of the company

`date`: in the format yyyy-mm-dd

`price`: the current price per stock at the datetime

Data source: Yahoo! Finance

Notes: Data was selected at 9:30 am each day (stock market open). This project was inspired by Codecademy's "Project: Watching the Stock Market"


***

### Step 1: Setup SQL connection

In [1]:
import sqlite3
import pandas as pd

connection = sqlite3.connect("stocks.db")
cursor = connection.cursor()

### Step 2: Create Table

In [2]:
#table creation
cursor.execute(''' CREATE TABLE jan_week_stocks (
symbol TEXT,
name TEXT,
date TEXT NOT NULL,
price FLOAT
);
''')

#IBM values
IBM_mon = ("IBM", "IBM Common Stock", "2022-01-24", 127.99)
IBM_tues = ("IBM", "IBM Common Stock", "2022-01-25", 129.14)
IBM_wed = ("IBM", "IBM Common Stock", "2022-01-26", 136.47)
IBM_thur = ("IBM", "IBM Common Stock", "2022-01-27", 133.66)
IBM_fri = ("IBM", "IBM Common Stock", "2022-01-28", 133.19)
IBM_final = ("IBM", "IBM Common Stock", "2022-01-31", 134.09)

#Intel values
INTC_mon = ("INTC", "Intel Corporation", "2022-01-24", 51.48)
INTC_tues = ("INTC", "Intel Corporation", "2022-01-25", 51.01)
INTC_wed = ("INTC", "Intel Corporation", "2022-01-26", 51.13)
INTC_thur = ("INTC", "Intel Corporation", "2022-01-27", 49.98)
INTC_fri = ("INTC", "Intel Corporation", "2022-01-28", 47.71)
INTC_final = ("INTC", "Intel Corporation", "2022-01-31", 47.68)

#insert IBM values
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_mon)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_tues)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_wed)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_thur)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_fri)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', IBM_final)

#insert Intel values
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_mon)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_tues)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_wed)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_thur)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_fri)
cursor.execute('INSERT INTO jan_week_stocks VALUES (?,?,?,?);', INTC_final)

#commit changes
connection.commit()

#print via pandas
df = pd.read_sql_query("SELECT * FROM jan_week_stocks", connection)
print(df)

   symbol               name        date   price
0     IBM   IBM Common Stock  2022-01-24  127.99
1     IBM   IBM Common Stock  2022-01-25  129.14
2     IBM   IBM Common Stock  2022-01-26  136.47
3     IBM   IBM Common Stock  2022-01-27  133.66
4     IBM   IBM Common Stock  2022-01-28  133.19
5     IBM   IBM Common Stock  2022-01-31  134.09
6    INTC  Intel Corporation  2022-01-24   51.48
7    INTC  Intel Corporation  2022-01-25   51.01
8    INTC  Intel Corporation  2022-01-26   51.13
9    INTC  Intel Corporation  2022-01-27   49.98
10   INTC  Intel Corporation  2022-01-28   47.71
11   INTC  Intel Corporation  2022-01-31   47.68


### Step 3: Distinct Stocks

In [8]:
#print distinct stocks
for row in cursor.execute('SELECT DISTINCT name FROM jan_week_stocks;'):
    print(row)

('IBM Common Stock',)
('Intel Corporation',)


### Step 4: Price Ranges

In [11]:
#Price above 130 (inclusive) w/ dates
cursor.execute('SELECT symbol, date, price FROM jan_week_stocks WHERE price >= 130.0;').fetchall()

[('IBM', '2022-01-26', 136.47),
 ('IBM', '2022-01-27', 133.66),
 ('IBM', '2022-01-28', 133.19),
 ('IBM', '2022-01-31', 134.09)]

In [12]:
#Price below 50 (inclusive) w/ dates
cursor.execute('SELECT symbol, date, price FROM jan_week_stocks WHERE price <= 50.0;').fetchall()

[('INTC', '2022-01-27', 49.98),
 ('INTC', '2022-01-28', 47.71),
 ('INTC', '2022-01-31', 47.68)]

### Step 5: Key Statistics

#### Minimum Prices

In [16]:
#Minimum price for IBM
cursor.execute('SELECT * FROM jan_week_stocks WHERE symbol = "IBM" ORDER BY price ASC LIMIT 1').fetchall()

[('IBM', 'IBM Common Stock', '2022-01-24', 127.99)]

In [17]:
#Minimum price for Intel
cursor.execute('SELECT * FROM jan_week_stocks WHERE symbol = "INTC" ORDER BY price ASC LIMIT 1').fetchall()

[('INTC', 'Intel Corporation', '2022-01-31', 47.68)]

#### Maximum Prices

In [18]:
#Maximum price for IBM
cursor.execute('SELECT * FROM jan_week_stocks WHERE symbol = "IBM" ORDER BY price DESC LIMIT 1').fetchall()

[('IBM', 'IBM Common Stock', '2022-01-26', 136.47)]

In [19]:
#Maximum price for Intel
cursor.execute('SELECT * FROM jan_week_stocks WHERE symbol = "INTC" ORDER BY price DESC LIMIT 1').fetchall()

[('INTC', 'Intel Corporation', '2022-01-24', 51.48)]

#### Average Prices

In [23]:
#Average price for IBM
cursor.execute('SELECT AVG(price) FROM jan_week_stocks WHERE symbol = "IBM";').fetchone()

(132.42333333333335,)

In [26]:
#Average price for Intel
cursor.execute('SELECT AVG(price) FROM jan_week_stocks WHERE symbol = "INTC";').fetchone()

(49.83166666666667,)

In [ ]:
connection.close()

## Conclusions

There are a number of key statistics that can be pulled from this analysis:

- The IBM stock went above 130 on open the following dates: 
    - 2022-01-26
    - 2022-01-27
    - 2022-01-28
    - 2022-01-31
- The intel stock dropped below 50 on open the following dates:
    - 2022-01-27
    - 2022-01-28
    - 2022-01-31
- The minimum and maximum stock price for IBM was 127.99 / 136.47
- The minimum and maximum stock price for Intel was 47.68 / 51.48
- The average stock price for IBM was 132.42
- The average stock price for Intel was 49.83